NBitcoin/TransactionBuilder.cs

```diff
@@ -1189,25 +1189,20 @@ public void Build(TransactionBuildingContext ctx)
 				{
 					var fee = ctx.CurrentGroupContext.Fee.GetAmount(Money.Zero);
 					txout.Value -= fee;
-
-					var minimumTxOutValue = (parent.DustPrevention ? parent.GetDust(txout.ScriptPubKey) : Money.Zero);
-					if (txout.Value < Money.Zero)
-					{
-						throw new OutputTooSmallException("Can't substract fee from this output because the amount is too small",
-						ctx.Group.Name,
-						-txout.Value
-						);
-					}
 					ctx.CurrentGroupContext.FeePaid = true;
-					if (txout.Value < minimumTxOutValue)
-					{
-						// Between zero and dust, should strip this output.
-						ctx.CurrentGroupContext.DustPreventionTotalRemoved += txout.Value;
-						return;
-					}
 					ctx.CurrentGroupContext.FeeTxOut = txout;
 				}
 
+				var minimumTxOutValue = (parent.DustPrevention ? parent.GetDust(txout.ScriptPubKey) : Money.Zero);
+				if (txout.Value < minimumTxOutValue)
+				{
+					// If the txout is below dust, they throw an exception
+					throw new OutputTooSmallException("Can't substract fee from this output because the amount is too small",

```

Comment:
this error is out of `if (SubstractFee` block. But it says `Can't substract fee` 

This error will be thrown even user tries to send w/o SubstractFee, if the amount is less than dust. Kinda mis-leading.